internal/backends/qt/qt_window.rs

```diff
@@ -106,6 +108,14 @@ cpp! {{
             });
         }
         void mouseReleaseEvent(QMouseEvent *event) override {
+            // HACK: Qt on windows is a bit special when clicking on the window close button.
+            //       In that case the closeEvent is triggered, followed by a release event that
+            //       did not have a press event. This confuses Slint, so eat this event.
```

Comment:
Your comment is more general, but unfortunately it does not describe what the code actually does! If you create a popup on the press event, then the isMouseButtonDown will be set and release event code will trigger. 

This hack only works when Qt sends a release without a previous press. The one case I am sure this happens is when clicking the close button.

Reply:
> If you create a popup on the press event, then the isMouseButtonDown will be set and release event code will trigger.

But it will be set on the main window, not on the popup. So if the release is sent to the popup it will not trigger.

> The one case I am sure this happens is when clicking the close button on a window running on Windows.

Are you sure you don't get a press in that case?

Reply:
After a quick look at the Qt code. My suspicion is that the release you are seeing should have been a `QEvent::NonClientAreaMouseButtonRelease` but was not understood this way because it was sent to another window than the `QEvent::NonClientAreaMouseButtonPressed`

Reply:
Anyway, I think the explanation should at least tell that this is for the case where a popup is created from the close event because of a click on the [x] in the title bar.
Because now the comment talk about `closeEvent` but for the reader of this code, it is not obvious at all why the `closeEvent` is related to clicks.
eg: Do you see the same behavior of release event if we do not open a popup but just cancel the close?

Reply:
Better now?

> Do you see the same behavior of release event if we do not open a popup but just cancel the close?

No, only when the close event gets ignored. I wonder whether that close event has some detail that this could be based on instead of this stupid heuristic.

Reply:
I believe this is just a bug in Qt